traits/ctraits.c

```diff
@@ -3094,34 +3094,20 @@ _trait_set_default_value(trait_object *trait, PyObject *args)
 }
 
 /*-----------------------------------------------------------------------------
-|  Get or set the 'default_value_type' and 'default_value' fields
-|  of a CTrait instance. Use of this function for setting the default
-|  value information is deprecated; use set_default_value instead.
+|  Get the 'default_value_type' and 'default_value' fields
+|  of a CTrait instance.
 +----------------------------------------------------------------------------*/
 
 static PyObject *
-_trait_default_value(trait_object *trait, PyObject *args)
+_trait_default_value(trait_object *trait, PyObject *Py_UNUSED(ignored))
```

Comment:
Do we need to define a second argument to the `_trait_default_value` function, which is then marked as ignored (IIUC this is done only to silence compiler warnings)?

Reply:
Yes, a second argument is required here. You can see lots of uses of this pattern in the CPython source, too. The function has to be of type `PyCFunction` (docs [here](https://docs.python.org/3/c-api/structures.html)), which is defined as
```
typedef PyObject *(*PyCFunction)(PyObject *, PyObject *);
```

Reply:
and yes, the `Py_UNUSED` is there to silence compiler warnings.

Reply:
Thanks for the reference and explanation. Still LGTM